doc/manual/style-guide.rst

```diff
@@ -155,6 +155,26 @@ In this case :func:`cell(n) <cell>` is better. It is also more helpful to the co
 to annotate specific uses (e.g. ``a[i]::Int``) than to try to pack many
 alternatives into one type.
 
+Prefer methods over direct field access
+---------------------------------------
+
+Idiomatic Julia code should treat a module's exported methods as the interface
+to its types. A type's fields are generally considered implementation details
+and user code should not be expected to access them directly. This has several
+benefits:
+
+- Package developers are free to change the implementation without breaking
+  user code
+- Methods can be passed to higher-order constructs like `map` (e.g.
+  `map(length, items))` rather than `[i.length for i in items]`)
```

Comment:
this feels like a strawman example, and the takeaway (for me) is more "don't use fields for things which could be methods".... personally I'm not sure I agree that direct field access should be discouraged in general.

BUT a better example is for complex numbers: use `imag(z)` over `z.im`. The point is they lower to the same for Complex, but you could use imag on a complex number say in polar form.